### PR TITLE
Move onRouteUpdate calls to use @reach navigate's promise to ensure navigation always is actually done & committed to DOM

### PR DESCRIPTION
**Pull request by KyleAMathews**
Ryan Florence says this is necessary for React Suspense.

### CODE REVIEW

File: packages/gatsby/cache-dir/navigation.js [hunk 1]
```diff
@@ -74,10 +74,13 @@ const navigate = (to, options) => {
     if (!pageResources && process.env.NODE_ENV === `production`) {
       loader.getResourcesForPathname(`/404.html`).then(() => {
         clearTimeout(timeoutId)
+        onPreRouteUpdate(window.location)
         reachNavigate(to, options)
+        reachNavigate(to, options).then(() => onRouteUpdate(window.location))
```
Comment (m-allanson) [inline packages/gatsby/cache-dir/navigation.js]: double `reachNavigate` call?

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6d5d30c6aed659e226cbae

**Comment by KyleAMathews** [comment]
Deploy preview for *using-styled-components* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/using-styled-components/deploys/5b6d5d31c6aed659e226cbba

File: packages/gatsby/cache-dir/navigation.js (hunk 1 above)
Comment (KyleAMathews) [reply packages/gatsby/cache-dir/navigation.js]: haha yes 🤦‍♂️

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/using-glamor/deploys/5b6d5d31c6aed659e226cbb7

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/using-contentful/deploys/5b6d5d31c6aed659e226cbbd

**Comment by KyleAMathews** [comment]
Deploy preview for *image-processing* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/image-processing/deploys/5b6d5d31c6aed659e226cbb4

**Comment by KyleAMathews** [comment]
Deploy preview for *using-styled-components* failed.

Built with commit 8308dc845dacac111248c8adb703f88570eca9b7

https://app.netlify.com/sites/using-styled-components/deploys/5b6d5daac965924aac3ae9b1

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/using-contentful/deploys/5b6d5e7667610c1d583702e8

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 8308dc845dacac111248c8adb703f88570eca9b7

https://app.netlify.com/sites/using-glamor/deploys/5b6d5daac965924aac3ae9ae

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6d5e7667610c1d583702d9

**Comment by KyleAMathews** [comment]
Deploy preview for *image-processing* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/image-processing/deploys/5b6d5e7667610c1d583702df

**Comment by KyleAMathews** [comment]
Deploy preview for *using-jss* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/using-jss/deploys/5b6d5e7667610c1d58370300

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/using-glamor/deploys/5b6d5e7667610c1d583702e2

**Comment by KyleAMathews** [comment]
Deploy preview for *using-styled-components* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/using-styled-components/deploys/5b6d5e7667610c1d583702e5

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit c6a02b2810fd3f5e381469b503add491eed5eddb

https://app.netlify.com/sites/gatsbyjs/deploys/5b6d5e7667610c1d583702ce

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 129316ce639d41d8978c891a33f96ff5a05fa086

https://app.netlify.com/sites/gatsbyjs/deploys/5b6d5d2fc6aed659e226cba3

**Pull request merged**